keras/callbacks.py

```diff
@@ -381,3 +388,85 @@ def __init__(self, schedule):
 
     def on_epoch_begin(self, epoch, logs={}):
         self.model.optimizer.lr.set_value(self.schedule(epoch))
+
+
+class TensorBoard(Callback):
+    ''' Tensorboard basic visualizations.
+
+    This callback writes a log usable with TensorBoard.
+    Tensorboard is a visualization tools provided with TensorFlow.
+
+    If you have installed TensorFlow with pip, you should be able
+    to launch TensorBoard from the command line:
+    ```
+    tensorboard --logdir=/full_path_to_your_logs
+    ```
+    You could find more information at:
+    https://www.tensorflow.org/versions/master/how_tos/summaries_and_tensorboard/index.html
+
+    # Arguments
+        model: a keras model linked to a tensorflow session
+        feed: a dictionnary mapping tensors (inputs, outputs, weigths)
+            from the model._test keras function i.e. model._test.inputs
+            to the corresponding arrays.
+        freq: the frequency at which the callback will output
+            parameters and metrics to the log
+        log_dir: the path of the directory where to save the log
+            files to be parsed by tensorboard
+    '''
+    def __init__(self, model, feed, freq=2, log_dir='./logs',
+                 show_accuracy=False):
+        super(Callback, self).__init__()
+        assert _BACKEND == 'tensorflow', \
+            'TensorBoard callback only works with the tensorflow backend'
+        import tensorflow as tf
+        import keras.backend.tensorflow_backend as KTF
+
+        self.model = model
+        self.freq = freq
+        self.log_dir = log_dir
+        self.sess = KTF._get_session()
+        self.feed = feed
+
+        mod_type = self.model.get_config()['name']
+        if mod_type == 'Sequential':
+            layers = {l.get_config()['name']: l for l in self.model.layers}
+        elif mod_type == 'Graph':
+            layers = self.model.nodes
+        else:
+            raise Exception('Unrecognized model:',
+                            self.model.get_config()['name'])
+        for l in layers:
+            cur_layer = layers[l]
+            if hasattr(cur_layer, 'W'):
+                tf.histogram_summary('{}_W'.format(l), cur_layer.W)
+            if hasattr(cur_layer, 'b'):
+                tf.histogram_summary('{}_b'.format(l), cur_layer.b)
+            if hasattr(cur_layer, 'get_output'):
+                tf.histogram_summary('{}_out'.format(l),
+                                     cur_layer.get_output())
+        if show_accuracy is True and mod_type == 'Sequential':
+            f_output = self.model._test_with_acc
+            tf.scalar_summary('Accuracy',
+                              f_output.outputs[1])
+            tf.scalar_summary('Loss',
+                              f_output.outputs[0])
+        else:
+            f_output = self.model._test
+            losses = [self.model.loss[loss] for loss in self.model.loss]
```

Comment:
This won't work for a Sequential model. I'd recommend separating the two cases (graph / sequential) entirely, for clarity.